pkg/operator/highcpuusagealertcontroller/highcpuusagealert_controller.go

```diff
@@ -0,0 +1,91 @@
+package highcpuusagealertcontroller
+
+import (
+	"bytes"
+	"context"
+	"strconv"
+	"time"
+
+	v1 "github.com/openshift/api/config/v1"
+	configv1informers "github.com/openshift/client-go/config/informers/externalversions/config/v1"
+	configlistersv1 "github.com/openshift/client-go/config/listers/config/v1"
+	"github.com/openshift/cluster-kube-apiserver-operator/bindata"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceread"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/dynamic"
+)
+
+type highCPUUsageController struct {
+	client               dynamic.Interface
+	infraLister          configlistersv1.InfrastructureLister
+	clusterversionLister configlistersv1.ClusterVersionLister
+}
+
+func NewHighCPUUsageAlertController(configInformer configv1informers.Interface, client dynamic.Interface, recorder events.Recorder) factory.Controller {
+	c := &highCPUUsageController{
+		client:               client,
+		infraLister:          configInformer.Infrastructures().Lister(),
+		clusterversionLister: configInformer.ClusterVersions().Lister(),
+	}
+
+	return factory.New().
+		WithInformers(configInformer.Infrastructures().Informer(), configInformer.ClusterVersions().Informer()).
+		WithSync(c.sync).ResyncEvery(3*time.Minute).
+		ToController("highCPUUsageController", recorder.WithComponentSuffix("high-cpu-usage-controller"))
+}
+
+func (c *highCPUUsageController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	infra, err := c.infraLister.Get("cluster")
+	if err != nil {
+		return err
+	}
+
+	// we care only about SNO clusters
+	if infra.Status.InfrastructureTopology != v1.SingleReplicaTopologyMode {
+		return nil
+	}
+
```

Comment:
By passing `EnabledCapabilities` to `snoAlert` we would have all preconditions listed and checked before reconciling the alert.